src/com/owncloud/android/ui/activity/Uploader.java

```diff
@@ -683,8 +744,137 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
         return retval;
     }
+    
+    private OCFile getCurrentFolder(){
+        OCFile file = mFile;
+        if (file != null) {
+            if (file.isFolder()) {
+                return file;
+            } else if (getStorageManager() != null) {
+                String parentPath = file.getRemotePath().substring(0, 
+                                                            file.getRemotePath().lastIndexOf(file.getFileName()));
```

Comment:
This seems to be a repetitive scenario, could you please add some utility function in OCFile for getting parent dir path?
Something like:
```java
  String getParentRemotePath() {
    return new File(getRemotePath()).getParent();
  }
```